SignalServiceKit/src/Storage/FullTextSearchFinder.swift

```diff
@@ -193,58 +193,58 @@ public class FullTextSearchFinder: NSObject {
         return "\(recipientId) \(nationalNumber) \(displayName)"
     }
 
-    private static let messageIndexer: SearchIndexer<TSMessage> = SearchIndexer { (message: TSMessage) in
+    private static let messageIndexer: SearchIndexer<TSMessage> = SearchIndexer { (message: TSMessage, transaction: YapDatabaseReadTransaction) in
         if let body = message.body, body.count > 0 {
             return body
         }
-        if let oversizeText = oversizeText(forMessage: message) {
+        if let oversizeText = oversizeText(forMessage: message, transaction: transaction) {
             return oversizeText
         }
         return ""
     }
 
-    private static func oversizeText(forMessage message: TSMessage) -> String? {
+    private static func oversizeText(forMessage message: TSMessage, transaction: YapDatabaseReadTransaction) -> String? {
         guard message.hasAttachments() else {
             return nil
         }
-        let dbConnection = OWSPrimaryStorage.shared().dbReadConnection
-        var oversizeText: String?
-        dbConnection.read({ (transaction) in
```

Comment:
In this update YapDB passes a transaction to the indexing method, so we now pass it through instead of building this sneaky transaction.

Reply:
Fantastic.

Reply:
Hopefully that fixes (or at least reduces) some of the inconsistency issues that we required the guards below.